### ISSUE TITLE
Create index - Connection reset by peer 

### ISSUE THREAD

**Issue by kevinzezel** [milvus-io/milvus#7460]
Hi,

After 8 hours creating index, i receave this message error:

Code:
```
default_index = {"index_type": "IVF_SQ8", "params": {"nlist": 16384}, "metric_type": "IP"} 
collection.create_index(field_name="embedding", index_params=default_index,timeout=2592000)
```

Error:
```
Addr [172.21.0.5:19530] get_index_state
RPC error: <_MultiThreadedRendezvous of RPC that terminated with:
	status = StatusCode.UNAVAILABLE
	details = "Connection reset by peer"
	debug_error_string = "{"created":"@1630611910.433450269","description":"Error received from peer ipv4:172.21.0.5:19530","file":"src/core/lib/surface/call.cc","file_line":1067,"grpc_message":"Connection reset by peer","grpc_status":14}"
```

```
Environment:
Milvus version: v2.0.0-RC5
Deployment mode: standalone
SDK version: pymilvus v2.0.0rc5
OS: Ubuntu 18.04.1-64 Minimal for Bare Metal
CPU/Memory: 2x 2.2GHz Intel Xeon-Broadwell (E5-2650-V4)/ 756gb ram
GPU: 2x NVIDIA Tesla-V100-PCIe-32GB
Others: IBM Bare Metal
Docker image Ubuntu 20.04 and Python 3.8.10
```

Thanks,
Kevin

**Comment by yanliang567** [comment]
@kevinzezel how many entities did you insert? Also could you please provide the logs for investigation?

**Comment by kevinzezel** [comment]
45MM

Error: <DescribeCollectionException: (code=1, message=proxy not healthy)>
Error: <DescribeCollectionException: (code=1, message=proxy not healthy)>
Exception in thread Thread-35:
Traceback (most recent call last):
  File "/usr/lib/python3.8/threading.py", line 932, in _bootstrap_inner
    self.run()
  File "/usr/lib/python3.8/threading.py", line 870, in run
    self._target(*self._args, **self._kwargs)
  File "subir_milvus.py", line 84, in subir
    mr = collection.insert(data=[aux1,step])
  File "/usr/local/lib/python3.8/dist-packages/pymilvus/orm/pymilvus_orm/collection.py", line 522, in insert
    res = conn.insert(collection_name=self._name, entities=entities, ids=None,
  File "/usr/local/lib/python3.8/dist-packages/pymilvus/client/stub.py", line 66, in handler
    raise e
  File "/usr/local/lib/python3.8/dist-packages/pymilvus/client/stub.py", line 52, in handler
    return func(self, *args, **kwargs)
  File "/usr/local/lib/python3.8/dist-packages/pymilvus/client/stub.py", line 40, in inner
    return func(self, *args, **kwargs)
  File "/usr/local/lib/python3.8/dist-packages/pymilvus/client/stub.py", line 859, in insert
    return handler.bulk_insert(collection_name, entities, partition_name, timeout, **kwargs)
  File "/usr/local/lib/python3.8/dist-packages/pymilvus/client/grpc_handler.py", line 67, in handler
    raise e
  File "/usr/local/lib/python3.8/dist-packages/pymilvus/client/grpc_handler.py", line 59, in handler
    return func(self, *args, **kwargs)
  File "/usr/local/lib/python3.8/dist-packages/pymilvus/client/grpc_handler.py", line 541, in bulk_insert
    raise err
  File "/usr/local/lib/python3.8/dist-packages/pymilvus/client/grpc_handler.py", line 527, in bulk_insert
    request = self._prepare_bulk_insert_request(collection_name, entities, partition_name, timeout, **kwargs)
  File "/usr/local/lib/python3.8/dist-packages/pymilvus/client/grpc_handler.py", line 510, in _prepare_bulk_insert_request
    collection_schema = self.describe_collection(collection_name, timeout=timeout, **kwargs)
  File "/usr/local/lib/python3.8/dist-packages/pymilvus/client/grpc_handler.py", line 67, in handler
    raise e
  File "/usr/local/lib/python3.8/dist-packages/pymilvus/client/grpc_handler.py", line 59, in handler
    return func(self, *args, **kwargs)
  File "/usr/local/lib/python3.8/dist-packages/pymilvus/client/grpc_handler.py", line 418, in describe_collection
    raise DescribeCollectionException(status.error_code, status.reason)

**Comment by yanliang567** [comment]
@sunby is it a know issue of insert deadlock?

**Comment by sunby** [comment]
> @sunby is it a know issue of insert deadlock?

I am not sure. @kevinzezel Can you provide the proxy log file?

**Comment by kevinzezel** [comment]
@sunby Where can i find the proxy log file?

**Comment by kevinzezel** [comment]
docker logs --tail 100 milvus-standalone

```
root@ib0500007-dti:~# docker logs --tail 100 milvus-standalone 
[2021/09/08 12:40:26.102 +00:00] [DEBUG] [etcd_kv.go:291] [MultiSaveAndRemove]
[2021/09/08 12:40:26.103 +00:00] [DEBUG] [move_binlog_path_helper.go:58] ["success to move binlogs of segment 427517577114157057"]
[2021/09/08 12:40:26.103 +00:00] [DEBUG] [etcd_kv.go:63] ["LoadWithPrefix "] [prefix=by-dev/meta/datacoord/binlog/segment/427516191604473858]
[2021/09/08 12:40:26.103 +00:00] [DEBUG] [etcd_kv.go:291] [MultiSaveAndRemove]
[2021/09/08 12:40:26.104 +00:00] [DEBUG] [move_binlog_path_helper.go:58] ["success to move binlogs of segment 427516191604473858"]
[2021/09/08 12:40:26.104 +00:00] [DEBUG] [etcd_kv.go:63] ["LoadWithPrefix "] [prefix=by-dev/meta/datacoord/binlog/segment/427517555618349057]
[2021/09/08 12:40:26.112 +00:00] [DEBUG] [etcd_kv.go:291] [MultiSaveAndRemove]
[2021/09/08 12:40:26.112 +00:00] [DEBUG] [move_binlog_path_helper.go:58] ["success to move binlogs of segment 427517555618349057"]
[2021/09/08 12:40:26.112 +00:00] [DEBUG] [etcd_kv.go:63] ["LoadWithPrefix "] [prefix=by-dev/meta/datacoord/binlog/segment/427517779856326658]
[2021/09/08 12:40:26.120 +00:00] [DEBUG] [etcd_kv.go:291] [MultiSaveAndRemove]
[2021/09/08 12:40:26.120 +00:00] [DEBUG] [move_binlog_path_helper.go:58] ["success to move binlogs of segment 427517779856326658"]
[2021/09/08 12:40:26.120 +00:00] [DEBUG] [etcd_kv.go:63] ["LoadWithPrefix "] [prefix=by-dev/meta/datacoord/binlog/segment/427515944350515201]
[2021/09/08 12:40:26.128 +00:00] [DEBUG] [etcd_kv.go:291] [MultiSaveAndRemove]
[2021/09/08 12:40:26.129 +00:00] [DEBUG] [move_binlog_path_helper.go:58] ["success to move binlogs of segment 427515944350515201"]
[2021/09/08 12:40:26.129 +00:00] [DEBUG] [etcd_kv.go:63] ["LoadWithPrefix "] [prefix=by-dev/meta/datacoord/binlog/segment/427516045970374657]
[2021/09/08 12:40:26.136 +00:00] [DEBUG] [etcd_kv.go:291] [MultiSaveAndRemove]
[2021/09/08 12:40:26.137 +00:00] [DEBUG] [move_binlog_path_helper.go:58] ["success to move binlogs of segment 427516045970374657"]
[2021/09/08 12:40:26.137 +00:00] [DEBUG] [etcd_kv.go:63] ["LoadWithPrefix "] [prefix=by-dev/meta/datacoord/binlog/segment/427517535669190657]
[2021/09/08 12:40:26.137 +00:00] [DEBUG] [etcd_kv.go:291] [MultiSaveAndRemove]
[2021/09/08 12:40:26.138 +00:00] [DEBUG] [move_binlog_path_helper.go:58] ["success to move binlogs of segment 427517535669190657"]
[2021/09/08 12:40:26.138 +00:00] [DEBUG] [server.go:214] ["dataCoordinator startup success"]
[2021/09/08 12:40:26.148 +00:00] [DEBUG] [proxy.go:114] ["Proxy dataCoord is ready"]
[2021/09/08 12:40:26.148 +00:00] [DEBUG] [proxy.go:119] ["Proxy wait for queryCoord ready"]
[2021/09/08 12:40:26.167 +00:00] [DEBUG] [proxy.go:125] ["Proxy queryCoord is ready"]
[2021/09/08 12:40:26.167 +00:00] [DEBUG] [proxy.go:130] ["Proxy wait for indexCoord ready"]
[2021/09/08 12:40:26.178 +00:00] [DEBUG] [index_coord.go:241] ["get IndexCoord component states ..."]
[2021/09/08 12:40:26.178 +00:00] [DEBUG] [index_coord.go:255] ["IndexCoord GetComponentStates"] ["IndexCoord component state"="nodeID:427584343150428161 role:\"IndexCoord\" state_code:Healthy "]
[2021/09/08 12:40:26.178 +00:00] [DEBUG] [proxy.go:136] ["Proxy indexCoord is ready"]
[2021/09/08 12:40:26.179 +00:00] [DEBUG] [meta.go:595] ["query coordinator create query channel"] [queryChannelName=search-0] [queryResultChannelName=searchResult-0]
[2021/09/08 12:40:26.179 +00:00] [DEBUG] [proxy.go:150] ["Proxy CreateQueryChannel success"]
[2021/09/08 12:40:26.179 +00:00] [DEBUG] [proxy.go:154] ["Proxy CreateQueryChannel success"] [SearchResultChannelNames="[searchResult-0]"]
[2021/09/08 12:40:26.179 +00:00] [DEBUG] [proxy.go:155] ["Proxy CreateQueryChannel success"] [RetrieveResultChannelNames="[searchResult-0]"]
[2021/09/08 12:40:26.180 +00:00] [DEBUG] [rocksmq_impl.go:165] ["RocksMQ: proxyTimeTick-0/begin_id or proxyTimeTick-0/end_id existed."]
[2021/09/08 12:40:26.180 +00:00] [DEBUG] [timetick.go:68] [proxy] ["proxy AsProducer"="[proxyTimeTick-0]"]
[2021/09/08 12:40:26.180 +00:00] [DEBUG] [service.go:119] ["proxy node init done ..."]
[2021/09/08 12:40:26.180 +00:00] [DEBUG] [proxy.go:344] ["init global meta cache ..."]
[2021/09/08 12:40:26.180 +00:00] [DEBUG] [proxy.go:347] ["start scheduler ..."]
[2021/09/08 12:40:26.180 +00:00] [DEBUG] [session_util.go:63] ["Session try to connect to etcd"]
[2021/09/08 12:40:26.181 +00:00] [DEBUG] [task_scheduler.go:666] [Proxy] [SearchResultChannelNames="[searchResult-0]"] [ProxySubName=proxy-239]
[2021/09/08 12:40:26.181 +00:00] [INFO] [grpclog.go:37] ["[core]parsed scheme: \"etcd-endpoints\""]
[2021/09/08 12:40:26.181 +00:00] [INFO] [grpclog.go:37] ["[core]ccResolverWrapper: sending update to cc: {[{etcd:2379 etcd <nil> 0 <nil>}] 0xc0034b8700 <nil>}"]
[2021/09/08 12:40:26.181 +00:00] [INFO] [grpclog.go:37] ["[core]ClientConn switching balancer to \"round_robin\""]
[2021/09/08 12:40:26.181 +00:00] [INFO] [grpclog.go:37] ["[core]Channel switches to new LB policy \"round_robin\""]
[2021/09/08 12:40:26.181 +00:00] [INFO] [grpclog.go:37] ["[balancer]base.baseBalancer: got new ClientConn state: {{[{etcd:2379 etcd <nil> 0 <nil>}] 0xc0034b8700 <nil>} <nil>}"]
[2021/09/08 12:40:26.181 +00:00] [INFO] [grpclog.go:37] ["[core]Subchannel Connectivity change to CONNECTING"]
[2021/09/08 12:40:26.181 +00:00] [INFO] [grpclog.go:37] ["[balancer]base.baseBalancer: handle SubConn state change: 0xc00349f5a0, CONNECTING"]
[2021/09/08 12:40:26.181 +00:00] [INFO] [grpclog.go:37] ["[core]Subchannel picks a new address \"etcd:2379\" to connect"]
[2021/09/08 12:40:26.181 +00:00] [INFO] [grpclog.go:37] ["[core]Channel Connectivity change to CONNECTING"]
[2021/09/08 12:40:26.182 +00:00] [INFO] [grpclog.go:37] ["[core]Subchannel Connectivity change to READY"]
[2021/09/08 12:40:26.182 +00:00] [INFO] [grpclog.go:37] ["[balancer]base.baseBalancer: handle SubConn state change: 0xc00349f5a0, READY"]
[2021/09/08 12:40:26.182 +00:00] [INFO] [grpclog.go:37] ["[roundrobin]roundrobinPicker: newPicker called with info: {map[0xc00349f5a0:{{etcd:2379 etcd <nil> 0 <nil>}}]}"]
[2021/09/08 12:40:26.182 +00:00] [INFO] [grpclog.go:37] ["[core]Channel Connectivity change to READY"]
[2021/09/08 12:40:26.183 +00:00] [DEBUG] [session_util.go:80] ["Sessiont connect to etcd success"]
[2021/09/08 12:40:26.183 +00:00] [DEBUG] [session_util.go:259] ["SessionUtil GetSessions"] [prefix=RootCoord] [resp="{\"header\":{\"cluster_id\":14841639068965178418,\"member_id\":10276657743932975437,\"revision\":275582,\"raft_term\":3},\"kvs\":[{\"key\":\"YnktZGV2L21ldGEvc2Vzc2lvbi9Sb290Q29vcmQ=\",\"create_revision\":275308,\"mod_revision\":275308,\"version\":1,\"value\":\"eyJTZXJ2ZXJJRCI6MjQxLCJTZXJ2ZXJOYW1lIjoiUm9vdENvb3JkIiwiQWRkcmVzcyI6ImxvY2FsaG9zdDo1MzEwMCIsIkV4Y2x1c2l2ZSI6dHJ1ZX0=\",\"lease\":7587857034672724273}],\"count\":1}"]
[2021/09/08 12:40:26.183 +00:00] [DEBUG] [client.go:58] ["RootCoordClient GetSessions success"]
[2021/09/08 12:40:26.183 +00:00] [DEBUG] [client.go:102] ["RootCoordClient try reconnect "] [address=localhost:53100]
[2021/09/08 12:40:26.183 +00:00] [INFO] [grpclog.go:37] ["[core]parsed scheme: \"\""]
[2021/09/08 12:40:26.183 +00:00] [INFO] [grpclog.go:37] ["[core]scheme \"\" not registered, fallback to default scheme"]
[2021/09/08 12:40:26.183 +00:00] [INFO] [grpclog.go:37] ["[core]ccResolverWrapper: sending update to cc: {[{localhost:53100  <nil> 0 <nil>}] <nil> <nil>}"]
[2021/09/08 12:40:26.183 +00:00] [INFO] [grpclog.go:37] ["[core]ClientConn switching balancer to \"pick_first\""]
[2021/09/08 12:40:26.184 +00:00] [INFO] [grpclog.go:37] ["[core]Channel switches to new LB policy \"pick_first\""]
[2021/09/08 12:40:26.184 +00:00] [INFO] [grpclog.go:37] ["[core]Subchannel Connectivity change to CONNECTING"]
[2021/09/08 12:40:26.184 +00:00] [INFO] [grpclog.go:37] ["[core]Subchannel picks a new address \"localhost:53100\" to connect"]
[2021/09/08 12:40:26.184 +00:00] [INFO] [grpclog.go:37] ["[core]pickfirstBalancer: UpdateSubConnState: 0xc00373c1b0, {CONNECTING <nil>}"]
[2021/09/08 12:40:26.184 +00:00] [INFO] [grpclog.go:37] ["[core]Channel Connectivity change to CONNECTING"]
[2021/09/08 12:40:26.184 +00:00] [INFO] [grpclog.go:37] ["[core]Subchannel Connectivity change to READY"]
[2021/09/08 12:40:26.184 +00:00] [INFO] [grpclog.go:37] ["[core]pickfirstBalancer: UpdateSubConnState: 0xc00373c1b0, {READY <nil>}"]
[2021/09/08 12:40:26.184 +00:00] [INFO] [grpclog.go:37] ["[core]Channel Connectivity change to READY"]
[2021/09/08 12:40:26.184 +00:00] [DEBUG] [client.go:138] ["RootCoordClient try reconnect success"]
[2021/09/08 12:40:26.184 +00:00] [DEBUG] [proxy.go:350] ["start id allocator ..."]
[2021/09/08 12:40:26.184 +00:00] [DEBUG] [proxy.go:353] ["start seg assigner ..."]
[2021/09/08 12:40:26.185 +00:00] [DEBUG] [proxy.go:356] ["start time tick ..."]
[2021/09/08 12:40:26.185 +00:00] [DEBUG] [proxy.go:362] ["start channelsTimeTicker"]
[2021/09/08 12:40:26.185 +00:00] [DEBUG] [proxy.go:372] [Proxy] [State=Healthy]
[2021/09/08 12:40:26.185 +00:00] [DEBUG] [service.go:124] ["proxy node start done ..."]
[2021/09/08 12:40:27.185 +00:00] [DEBUG] [segment.go:182] ["Proxy SegIDAssigner pickCanDoFunc"] [len(ToDoReqs)=0]
[2021/09/08 12:40:27.185 +00:00] [DEBUG] [segment.go:182] ["Proxy SegIDAssigner pickCanDoFunc"] [len(ToDoReqs)=0]
[2021/09/08 12:40:27.653 +00:00] [DEBUG] [session_util.go:259] ["SessionUtil GetSessions"] [prefix=IndexNode] [resp="{\"header\":{\"cluster_id\":14841639068965178418,\"member_id\":10276657743932975437,\"revision\":275583,\"raft_term\":3},\"kvs\":[{\"key\":\"YnktZGV2L21ldGEvc2Vzc2lvbi9JbmRleE5vZGUtMjQ0\",\"create_revision\":275313,\"mod_revision\":275313,\"version\":1,\"value\":\"eyJTZXJ2ZXJJRCI6MjQ0LCJTZXJ2ZXJOYW1lIjoiSW5kZXhOb2RlIiwiQWRkcmVzcyI6IjE3Mi4yMS4wLjY6MjExMjEifQ==\",\"lease\":7587857034672724288}],\"count\":1}"]
[2021/09/08 12:40:27.653 +00:00] [DEBUG] [index_coord.go:694] ["IndexCoord assignTaskLoop"] ["Available IndexNode IDs"="[244]"]
[2021/09/08 12:40:27.653 +00:00] [DEBUG] [index_coord.go:699] ["IndexCoord assignTaskLoop"] ["Unassigned tasks number"=0] ["Unassigned tasks meta"=null]
[2021/09/08 12:40:28.185 +00:00] [DEBUG] [segment.go:182] ["Proxy SegIDAssigner pickCanDoFunc"] [len(ToDoReqs)=0]
[2021/09/08 12:40:28.185 +00:00] [DEBUG] [segment.go:182] ["Proxy SegIDAssigner pickCanDoFunc"] [len(ToDoReqs)=0]
[2021/09/08 12:40:29.185 +00:00] [DEBUG] [segment.go:182] ["Proxy SegIDAssigner pickCanDoFunc"] [len(ToDoReqs)=0]
[2021/09/08 12:40:29.185 +00:00] [DEBUG] [segment.go:182] ["Proxy SegIDAssigner pickCanDoFunc"] [len(ToDoReqs)=0]
[2021/09/08 12:40:30.185 +00:00] [DEBUG] [segment.go:182] ["Proxy SegIDAssigner pickCanDoFunc"] [len(ToDoReqs)=0]
[2021/09/08 12:40:30.185 +00:00] [DEBUG] [segment.go:182] ["Proxy SegIDAssigner pickCanDoFunc"] [len(ToDoReqs)=0]
[2021/09/08 12:40:30.368 +00:00] [DEBUG] [rocksmq_retention.go:264] ["In ticker: "] [ticker=1631104830]
[2021/09/08 12:40:30.369 +00:00] [DEBUG] [rocksmq_retention.go:353] ["In expiredCleanUp: "] [topic=datacoord-statistics-channel] [endID=427515511636920821] [deletedAckedSize=90206999793]
[2021/09/08 12:40:30.369 +00:00] [DEBUG] [rocksmq_retention.go:403] ["Delete page info"] [topic=datacoord-statistics-channel] [pageStartID=427515511636516324] [pageEndID=427515511636920821]
panic: Invalid argument: end key comes before start key

goroutine 159 [running]:
github.com/milvus-io/milvus/internal/util/rocksmq/server/rocksmq.(*retentionInfo).retention.func1(0x2264640, 0xc00014a180, 0x2263b40, 0xc0076c3280, 0x1)
	/go/src/github.com/milvus-io/milvus/internal/util/rocksmq/server/rocksmq/rocksmq_retention.go:269 +0xb9
sync.(*Map).Range(0xc0002745f8, 0xc02eb8ff08)
	/usr/local/go/src/sync/map.go:345 +0x1d9
github.com/milvus-io/milvus/internal/util/rocksmq/server/rocksmq.(*retentionInfo).retention(0xc000274580, 0xc00013410f, 0xe)
	/go/src/github.com/milvus-io/milvus/internal/util/rocksmq/server/rocksmq/rocksmq_retention.go:265 +0x13e
created by github.com/milvus-io/milvus/internal/util/rocksmq/server/rocksmq.(*retentionInfo).startRetentionInfo
	/go/src/github.com/milvus-io/milvus/internal/util/rocksmq/server/rocksmq/rocksmq_retention.go:121 +0x11e

```

**Comment by fishpenguin** [comment]
This is a bug of rocksmq retention, and already been fixed in latest master branch by [pr7403](https://github.com/milvus-io/milvus/pull/7403/files) This will be released in 2.0.0-RC6.
